symsrv-fetch.py

```diff
@@ -24,345 +24,468 @@
+                    data = await resp.read()
+                    typ = get_type(data)
+                    if typ == "unknown":
+                        # try again
+                        await asyncio.sleep(0.5)
+                    elif typ == "pdb-v2":
+                        # too old: skip it
+                        log.debug(f"PDB v2 (skipped because too old): {url}")
+                        return None
                     else:
-                        log.warn("Couldn't fetch binary for %s, %s",
-                                 code_id, code_file)
-                        raise Win64ProcessError()
-                    continue
+                        return data
                 else:
-                    raise Win64ProcessError()
-            return None
+                    log.error(f"Cannot get data (status {resp.status}) for {url}: ")
+        except Exception as e:
+            log.debug(f"Error with {url}")
+            log.exception(e)
+            await asyncio.sleep(0.5)
 
-
-def server_has_file(filename):
-    '''
-    Send the symbol server a HEAD request to see if it has this symbol file.
-    '''
-    try:
-        r = requests.head(
-            urlparse.urljoin(
-                MOZILLA_SYMBOL_SERVER,
-                urllib.quote(filename)))
-        return r.status_code == 200
-    except requests.exceptions.RequestException:
-        return False
+    log.debug(f"Too much retries (GET) for {url}: give up.")
+    return None
 
 
 def write_skiplist(skiplist):
-    try:
-        with open(os.path.join(thisdir, 'skiplist.txt'), 'w') as sf:
-            for (debug_id, debug_file) in skiplist.iteritems():
-                sf.write('%s %s\n' % (debug_id, debug_file))
-    except IOError:
-        log.exception('Error writing skiplist.txt')
+    with open(os.path.join(thisdir, "skiplist.txt"), "w") as sf:
+        for (debug_id, debug_file) in skiplist.items():
+            sf.write("%s %s\n" % (debug_id, debug_file))
 
 
-def fetch_missing_symbols(u):
-    log.info('Trying missing symbols from %s' % u)
-    r = requests.get(u)
-    r.raise_for_status()
-    return r.text
+async def fetch_missing_symbols(u):
+    log.info("Trying missing symbols from %s" % u)
+    async with ClientSession() as client:
+        async with client.get(u, headers=HEADERS) as resp:
+            # just skip the first line since it contains column headers
+            data = await resp.text()
+            return data.splitlines()[1:]
 
 
-def main():
-    parser = argparse.ArgumentParser(
-        description='Fetch missing symbols from Microsoft symbol server')
-    parser.add_argument('--missing-symbols', type=str, help='missing symbols URL',
-                        default=MISSING_SYMBOLS_URL)
-    parser.add_argument('zip', type=str, help='output zip file')
-    args = parser.parse_args()
+async def get_list(filename, info):
+    alist = set()
+    path = os.path.join(thisdir, filename)
+    if os.path.exists(path):
```

Comment:
Nit: this will block all asyncio tasks. While we could transform it to async, the best solution here is to simply avoid the I/O to check if the file exist and just try to read it and catch and ignore the FileNotFoundError exception.